src/json-validation.ts

```diff
@@ -0,0 +1,40 @@
+import Ajv, { ValidateFunction } from "ajv";
+
+import { Instructions, RepoArray } from "./inputs";
+import instructionsSchema from "./json-schemas/Instructions.json";
+import repoArraySchema from "./json-schemas/RepoArray.json";
+
+type SchemaTypes = {
+  repoArray: RepoArray;
+  instructions: Instructions;
+};
+export type Schema = keyof SchemaTypes;
+
+const ajv = new Ajv();
+const validators: Record<Schema, () => ValidateFunction> = {
+  repoArray: memoize(() => ajv.compile(repoArraySchema)),
```

Comment:
Have you tested how fast it actually is to compile a schema? Could it be that compiling the schema is fast enough that we don't need to do this?

Reply:
I don't have precise timing data, but anecdotally it is very fast. Fast enough that it appears to return instantly. So I'm not terribly worried about it.

I went with writing the `memoize` function because of the desire to do it "properly" but you're right it's an unnoticeable performance gain if anything. So let's get rid of this and make the code simpler.